src/NServiceBus.Core.Tests/Serializers/SystemTextJson/SystemJsonSerializerTests.cs

```diff
@@ -221,6 +222,25 @@ public void Should_preserve_timezones()
                 Assert.AreEqual(expectedDateTimeOffsetUtc.Offset, result.DateTimeOffsetUtc.Offset);
             }
         }
+
+        [Test]
+        public void Should_be_able_to_serialize_payloads_with_BOM()
+        {
+            var payload = JsonSerializer.Serialize(new SomeMessage());
+
+            using (var stream = new MemoryStream())
+            {
+                using (var sw = new StreamWriter(stream, new UTF8Encoding(true)))
```

Comment:
For code readability to understand what `true` does.
```suggestion
                using (var sw = new StreamWriter(stream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: true)))
```